src/Widgets/Helper/Factories/Settings/ValueListFactory.php

```diff
@@ -0,0 +1,21 @@
+<?php
+
+
+namespace Ceres\Widgets\Helper\Factories\Settings;
+
+
+class ValueListFactory
+{
+    private $valueList;
+
+    public function addEntry($key, $caption)
+    {
+        $this->valueList[] = [$key => $caption];
```

Comment:
1. Du baust dir das Array an der Stelle falsch zusammen. Du baust gerade sowas:
```
[
  ["key1" => "caption1"],
  ["key2" => "caption2"]
]
```

was du möchtest:
```
[
  ["key" => "key1", "caption" => "caption1"],
  ["key" => "key2", "caption" => "caption2"]
]
```

Außerdem würde ich nicht $value in $caption umbenennen, sondern $key.

In Summe sollte die Funktion meiner Meinung nach so aussehen:
```
public function addEntry($caption, $value)
{
  $this->valueList[] = [
    "caption" => $caption,
    "value" => $value
  ];

  return $this;
}
```